lib/bandit/headers.ex

```diff
@@ -11,6 +11,22 @@ defmodule Bandit.Headers do
     end
   end
 
+  # covers ipv6 addresses, which look like this: `[::1]:4000`
+  def parse_hostlike_header("[" <> _ = host_header) do
+    host_header
+    |> :binary.split("]:")
+    |> case do
+      [host] ->
+        {:ok, host, nil}
+
+      [host, port] ->
+        case Integer.parse(port) do
+          {port, ""} when port > 0 -> {:ok, host <> "]", port}
+          _ -> {:error, "Header contains invalid port"}
+        end
```

Comment:
Suggest to reorder these clauses so they match those in the clause just below:

```suggestion
      [host, port] ->
        case Integer.parse(port) do
          {port, ""} when port > 0 -> {:ok, host <> "]", port}
          _ -> {:error, "Header contains invalid port"}
        end
        
      [host] ->
        {:ok, host, nil}        
```